### PR TITLE
Fix constant session regeneration

### PR DESCRIPTION
**Pull request by bizley**
| Q             | A
| ------------- | ---
| Is bugfix?    | ✔️
| New feature?  | ❌
| Breaks BC?    | ❌
| Fixed issues  | #19067

### CODE REVIEW

**Comment by samdark** [comment]
@bizley is the issue reproduce-able as a unit test?

**Comment by bizley** [comment]
@samdark I have removed the test environment condition for forced session regeneration on identity switching ( https://github.com/yiisoft/yii2/pull/19113/files#diff-c9b74b3b5811f9a082773e99476b87a3d7dd1f3d498323b26bb85208318f6ac3L658-L660 ) - was there is a valid reason for that? With that condition session ID testing is impossible without mocking the user component, so let me know.

**Comment by samdark** [comment]
@bizley that condition might be for Codeception module. I'm not sure. It was added by @qiangxue back in 2013.

**Comment by mikehaertl** [comment]
@bizley Sorry, my fix for #18646  messed this up. I added the call to `switchIdentity()` to get rid of stale identity data if the user is no longer logged in. Actually all I wanted is this:

https://github.com/yiisoft/yii2/blob/6fb16bc30f2efb63e9033d89d7d7a60d663480f8/framework/web/User.php#L661-L663

Would it make sense to move these into their own method (e.g. `removeIdentityData()`) and then call this from `renewAuthStatus()` and `switchIdentity()` instead?

**Comment by bizley** [comment]
@mikehaertl what about the approach I did here? I'm open to suggestions :)

File: framework/web/User.php [hunk 1]
```diff
@@ -655,9 +655,7 @@ public function switchIdentity($identity, $duration = 0)
         }
 
         $session = Yii::$app->getSession();
-        if (!YII_ENV_TEST) {
```
Comment (samdark) [inline framework/web/User.php:658]: I'd not touch it. Likely to break Codeception integration.

Comment (samdark) [reply framework/web/User.php:658]: Hmm... it's used in tests :)

**Review by samdark — APPROVED** [review]

File: framework/web/User.php [hunk 2]
```diff
@@ -655,9 +655,7 @@ public function switchIdentity($identity, $duration = 0)
...
-            $session->regenerateID(true);
-        }
+        $session->regenerateID(true);
         $session->remove($this->idParam);
         $session->remove($this->authTimeoutParam);
         $session->remove($this->authKeyParam);
```
Comment (samdark) [inline framework/web/User.php:661]: These might be unneeded now.

**Comment by samdark** [comment]
@mikehaertl that won't be enough to clean up all the leftovers including cookies.

**Comment by samdark** [comment]
Thanks for handling it, @bizley.

**Pull request merged**